test/k8sT/DatapathConfiguration.go

```diff
@@ -184,23 +195,35 @@ var _ = Describe("K8sDatapathConfig", func() {
 				"global.bpf.monitorFlags":       "psh",
 				"global.debug.enabled":          "false",
 			})
-			monitorOutput, _ := monitorConnectivityAcrossNodes(kubectl, monitorLog)
-
-			By("Checking the set of TCP notifications received matches expectations")
-			// | TCP Flags | Direction | Report? | Why?
-			// +===========+===========+=========+=====
-			// | SYN       |    ->     |    Y    | monitorAggregation=medium
-			// | SYN / ACK |    <-     |    Y    | monitorAggregation=medium
-			// | ACK       |    ->     |    N    | monitorFlags=(!ACK)
-			// | ACK       |    ...    |    N    | monitorFlags=(!ACK)
-			// | PSH       |    ->     |    Y    | monitorFlags=(PSH)
-			// | PSH       |    <-     |    Y    | monitorFlags=(PSH)
-			// | FIN       |    ->     |    Y    | monitorAggregation=medium
-			// | FIN / ACK |    <-     |    Y    | monitorAggregation=medium
-			// | ACK       |    ->     |    Y    | monitorAggregation=medium
-			egressPktCount := 4
-			ingressPktCount := 3
-			checkMonitorOutput(monitorOutput, egressPktCount, ingressPktCount)
+			monitorRes, monitorCancel, _ := monitorConnectivityAcrossNodes(kubectl)
+
+			var monitorOutput []byte
+			body := func() bool {
+				monitorOutput = monitorRes.CombineOutput().Bytes()
+
+				By("Checking the set of TCP notifications received matches expectations")
+				// | TCP Flags | Direction | Report? | Why?
+				// +===========+===========+=========+=====
+				// | SYN       |    ->     |    Y    | monitorAggregation=medium
+				// | SYN / ACK |    <-     |    Y    | monitorAggregation=medium
+				// | ACK       |    ->     |    N    | monitorFlags=(!ACK)
+				// | ACK       |    ...    |    N    | monitorFlags=(!ACK)
+				// | PSH       |    ->     |    Y    | monitorFlags=(PSH)
+				// | PSH       |    <-     |    Y    | monitorFlags=(PSH)
+				// | FIN       |    ->     |    Y    | monitorAggregation=medium
+				// | FIN / ACK |    <-     |    Y    | monitorAggregation=medium
+				// | ACK       |    ->     |    Y    | monitorAggregation=medium
+				egressPktCount := 4
+				ingressPktCount := 3
+				checkMonitorOutput(monitorOutput, egressPktCount, ingressPktCount)
+
+				return true
+			}
+
+			err := helpers.WithTimeout(body, "monitor aggregation did not send notifications", &helpers.TimeoutConfig{Timeout: helpers.HelperTimeout})
+			Expect(err).To(BeNil(), "Could not read monitor log")
+			monitorCancel()
```

Comment:
monitorCancel won't be run if the `Expect(err)` fails one line above.

Reply:
Good catch. I remember thinking that I had to do it this way but I can't see why now (maybe I though it was created inside the body function? clearly it isn't #codingmysteries).